pi5/pinctrl_utils.go

```diff
@@ -286,8 +289,88 @@ func (b *pinctrlpi5) createGPIOVPage(memPath string) error {
 	return err
 }
 
-// Cleans up mapped memory / files upon board close() call
-func (b *pinctrlpi5) cleanupPinControlMemory() error {
+/*
+For all Pins belonging to the same bank, pin data is stored contiguously and in 8 byte chunks.
+For a given pin, this method determines:
+ 1. which bank the pin belongs to
+ 2. the starting address of its 8 byte data chunk
+*/
+func getPinAddressOffset(pinNumber int) (int64, error) {
+
+	bankHeaderSize := 8 // 8 bytes of either header data assosciated with a bank (unsure about what is stored here though)
+
+	if !(1 <= pinNumber && pinNumber <= maxPinNum) {
+		return -1, errors.New("pin is out of bank range")
+	}
+
+	for i := 0; i < len(bankDivisions)-1; i++ {
+		if bankDivisions[i] <= pinNumber && pinNumber < bankDivisions[i+1] {
+
+			bankNum := i
+			bankBaseAddr := fselBankOffsets[bankNum]
+			pinBankOffset := pinNumber - bankDivisions[i]
+
+			pinAddressOffset := bankBaseAddr + bankHeaderSize + ((pinBankOffset) * fselPinDataSize)
+			return int64(pinAddressOffset), nil
+		}
+	}
+
+	return -1, errors.New("pin in bank range but not set")
+}
+
+// This method updates the given mode of a pin by finding its specific location in memory & writing to the 'mode' byte in the 8 byte block of pin data.
+func (b *pinctrlpi5) setPin(pinNumber int, newMode byte) error {
+
+	// Of the 8 bytes that represent a given pin's data, only the 4th index corresponds to the alternative mode setting
+	altModeIndex := 4
```

Comment:
4 bytes are allocated for alternative modes. However, you only need 1 byte to represent all the modes because 
1 byte = 8 bits
8 modes can be represented by 3 bits 

so in practice, only 1 byte is used and the rest are just 0xff's. the index of that byte in the entire chunk of pin data is byte[4].